src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/Transaction.java

```diff
@@ -0,0 +1,72 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.List;
+import java.util.Optional;
+
+public class Transaction extends Section {
+    @Getter(lazy = true)
+    private final ReferenceTransactionNumber referenceTransactionNumber =
+        ReferenceTransactionNumber.fromString(extractSegment(ReferenceTransactionNumber.KEY_QUALIFIER));
+    @Getter(lazy = true)
+    private final GpNameAndAddress gpNameAndAddress =
+        GpNameAndAddress.fromString(extractSegment(GpNameAndAddress.KEY_QUALIFIER));
+    @Getter(lazy = true)
+    private final Optional<PersonName> personName =
+        extractOptionalSegment(PersonName.KEY_QUALIFIER).map(PersonName::fromString);
```

Comment:
So, normally I'd want to insist that fields not be `Optional`s, but given the lombok weaving, this might be okay. On the other hand, I don't think these getters really need to be lazy - they're not so expensive.

Just a conversation point; ignore if uninteresting.